src/frontend/src/optimizer/plan_node/generic/agg.rs

```diff
@@ -418,7 +418,8 @@ impl<PlanRef: stream::StreamPlanRef> Agg<PlanRef> {
                 agg_kinds::single_value_state_iff_in_append_only!() if in_append_only => {
                     AggCallState::Value
                 }
-                agg_kinds::single_value_state!() => AggCallState::Value,
+                agg_kinds::single_value_state!()
+                | AggKind::Builtin(PbAggKind::ApproxPercentile) => AggCallState::Value,
```

Comment:
Shall we move the ApproxPercentile into the macro?

Reply:
Fixed https://github.com/risingwavelabs/risingwave/pull/17814/commits/7ca78e5081ebdb8b39619797d5079a31aa15c48e

Reply:
Actually it's not quite precise... `ApproxPercentile` is only a single value state when it's single phase agg.

Otherwise it doesn't quite fall under this classification.

Reply:
Because it will use custom executor instead.